parser-typechecker/src/Unison/Server/Backend.hs

```diff
@@ -810,9 +828,53 @@ renderDoc ppe width rt codebase r = do
         Nothing -> pure ()
       pure $ r <&> Term.amap (const mempty)
 
-    decls (Reference.DerivedId r) = fmap (DD.amap (const ())) <$> lift (Codebase.getTypeDeclaration codebase r)
+    decls (Reference.DerivedId r) = fmap (DD.amap (const ())) <$> Codebase.getTypeDeclaration codebase r
     decls _ = pure Nothing
 
+docsInBranchToHtmlFiles
+  :: Var v
+  => Rt.Runtime v
+  -> Codebase IO v Ann
+  -> Branch IO
+  -> FilePath
+  -> IO ()
+docsInBranchToHtmlFiles runtime codebase currentBranch directory = do
+  let deepTermRefs = (toList . R.range . Branch.deepTerms . Branch.head) currentBranch >>= toList
+  docRefs <- filterM (isDoc codebase) deepTermRefs
+  hqLength <- Codebase.hashLength codebase
+  let printNames = getCurrentPrettyNames (AllNames Path.empty) currentBranch
+  let width = defaultWidth
+  let ppe = PPE.fromNamesDecl hqLength printNames
+  docs <- concat <$> for docRefs (renderDoc ppe width runtime codebase . Referent.toReference)
+  liftIO $ traverse_ (renderDocToHtmlFile directory) docs
+
+  where
+    cleanPath :: FilePath -> FilePath
+    cleanPath filePath =
+      filePath <&> \case
+        '#' -> '@'
```

Comment:
We prefer `@` in urls (since `#` is reserved for url fragments), so we replace `#` to `@` to get a clean filename.